### ISSUE TITLE
border-*-width doesn't have default value

### ISSUE THREAD

**Issue by hrj** [UprootLabs/gngr#14]
For the following style,

```
div {
  border-bottom-style: solid;
  border-bottom-color: green;
}
```

A border-bottom-width of 1px should be assumed.

Currently, it becomes 0px.

Need to check where the default needs to be decided (css parser, render-state or layout)

**Comment by hrj** [comment]
Test case [here](https://uprootlabs.github.io/gngrTestCases/test-case/i14/)

jStyleParser is not inferring the `border-*-width` property. Will report upstream.

**Comment by hrj** [comment]
Added a comment to https://github.com/radkovo/jStyleParser/issues/16

**Comment by hrj** [comment]
There is no response from upstream yet. We can implement this in `gngr` itself.

@bogas04 Do you want to take a shot at it?

It requires changes in the class `HtmlValues`. It has a method called `getBorderInfo` which needs to automatically infer missing values for border-width, when other border values are specified.

**Comment by hrj** [comment]
On second thoughts, from a design perspective, it might be cleaner to do the inference in the caller of `getBorderInfo` which is `StyleSheetRenderState`.

**Comment by bogas04** [comment]
Sure! Thanks for the location, will work on it tonight.

**Comment by bogas04** [comment]
I was looking where the default border width was defined, and I came across [this](https://github.com/UprootLabs/gngr/blob/a6d6da31ea24d0c7cfe3f20c3928b87025ee6942/src/HTML_Renderer/org/lobobrowser/html/style/HtmlValues.java#L160-L170) commented part of the code. Do you think it was intended to serve the same purpose ?

**Comment by hrj** [comment]
@bogas04 Yes, that might be relevant code. To be extra sure, you can try running the tests from [grinder](https://github.com/UprootLabs/grinder). I am going offline now but will be back later in the night, if you face any problems.

**Comment by hrj** [comment]
I fixed this in https://github.com/UprootLabs/gngr/commit/f1db56b356447a3b8d0c2f14b2a35698d46ca358

Didn't wait because it was affecting a lot of tests (44).

**Comment by bogas04** [comment]
Sorry for being inactive, final sem exams upcoming and the old code wasn't keeping me interested enough.

**Comment by hrj** [comment]
@bogas04 No worries...

> and the old code wasn't keeping me interested enough.

LOL !

**Comment by hrj** [comment]
I am reopening this, because upstream replied after our fix and perhaps a different solution can be implemented.
